packages/keyring-controller/src/KeyringController.test.ts

```diff
@@ -2777,6 +2921,33 @@ describe('KeyringController', () => {
             );
           });
 
+        !cacheEncryptionKey &&
+          it('should not upgrade the vault encryption if the generic encryptor has the same parameters', async () => {
+            await withController(
+              {
+                skipVaultCreation: true,
+                cacheEncryptionKey,
+                state: { vault: 'my vault' },
+              },
+              async ({ controller, encryptor }) => {
+                jest.spyOn(encryptor, 'isVaultUpdated').mockReturnValue(true);
+                const encryptSpy = jest.spyOn(encryptor, 'encrypt');
+                jest.spyOn(encryptor, 'decrypt').mockResolvedValueOnce([
+                  {
+                    type: KeyringTypes.hd,
+                    data: {
+                      accounts: ['0x123'],
+                    },
+                  },
+                ]);
+
+                await controller.submitPassword(password);
+
+                expect(encryptSpy).toHaveBeenCalledTimes(1);
```

Comment:
I am not following, how does this show that the vault was not upgraded?

Reply:
I added this test before changing the implementation, and then I only changed the title. The vault in this test was being updated because of the missing metadata in the mock state. Fixed in [7a8953a](https://github.com/MetaMask/core/pull/5725/commits/7a8953a40b75b25d934262e36e514fee693b8b76)